components/function-controller/internal/controllers/serverless/function_reconcile.go

```diff
@@ -81,6 +83,10 @@ func (r *FunctionReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	if IsHealthCheckRequest(request) {
+		r.healthCh <- true
```

Comment:
If you would add `context.WithTimeout` please handle it here to not block routine because of overflowed channel